File: 2_Python/Algorithm/Sorting/bubblesort.py
def bubble_sort(array):
    """
    Bubble sort implementation

    :param array: unsorted array
    :return: None
    """

    size = len(array)

    for i in range(size):

        # Last i elements are already in place
        for j in range(size-i-1):

            if array[j] > array[j+1]:

                array[j], array[j+1] = array[j+1], array[j]

File: 4_Javascript/algorithm/sorting/bubble-sort.js
/**
 * Bubble sort implementation
 * @param {T[]} array - Unsorted array
 * @author Areeb Malik
 */
function bubbleSort(array) {
    let size = array.length;
    for(let i=0; i<size; i++) {
        // Last i elements are already in place
        for(let j=0; j<size-i-1; j++) {
            if(array[j] > array[j+1]) {
                let temp = array[j];
                array[j] = array[j+1];
                array[j+1] = temp;
            }
        }
    }
}
